libbeat/template/template.go

```diff
@@ -192,3 +211,34 @@ func (t *Template) generate(properties common.MapStr, dynamicTemplates []common.
 
 	return basicStructure
 }
+
+func loadYamlFile(path string) (common.Fields, error) {
+
+	cfg, err := yaml.NewConfigWithFile(path)
+	if err != nil {
+		return nil, err
+	}
+	return load(cfg)
+}
+
+func loadYamlByte(data []byte) (common.Fields, error) {
+
+	cfg, err := yaml.NewConfig(data)
+	if err != nil {
+		return nil, err
+	}
+	return load(cfg)
+}
+
+func load(cfg *ucfg.Config) (common.Fields, error) {
+	keys := []common.Field{}
```

Comment:
If you use `var keys common.Field` will `Unpack` allocate this for you?

Reply:
tested and seems to work.